src/bgg_companion_api.py

```diff
@@ -67,17 +74,34 @@ def get_object_ids(user: str) -> list[str]:
         return id_list
 
 
-def get_random_game(user: str) -> Union[BoardGame, UserIsNoneError]:
-    try:
-        id_list = get_object_ids(user)
-    except UserIsNoneError as exc:
-        return exc
-    id_string = ",".join(id_list)
-    board_games = get_board_games(id_string)
-    if board_games is None:
-        raise BoardGameIsNoneError("A board game was passed that does not exist within BoardGameGeek.")
+def game_type(game: BoardGame, gamefilter: GameFilter) -> bool:
+    return game.type == gamefilter.gameType
+
+
+def maxplayers_gt(game: BoardGame, gamefilter: GameFilter) -> bool:
+    return gamefilter.maxplayers is None or game.maxplayers >= gamefilter.maxplayers
+
+
+def exactplayers(game: BoardGame, gamefilter: GameFilter) -> bool:
+    return gamefilter.exactplayers is None or (game.maxplayers == gamefilter.exactplayers and game.minplayers == gamefilter.exactplayers)
+
+
+def game_matches_filter(game: BoardGame, gamefilter: GameFilter) -> bool:
+    return game_type(game=game, gamefilter=gamefilter) and \
+           maxplayers_gt(game=game, gamefilter=gamefilter) and \
+           exactplayers(game=game, gamefilter=gamefilter)
+
+
+def filter_games(board_games: list[BoardGame], game_filter: GameFilter) -> list[BoardGame]:
```

Comment:
I wonder if all these functions for filtering should actually be in their own module as they feel like they should be grouped together sharing the same responsibility?

Reply:
I'm thinking the same as you; personally once a module starts incorporating different functionality (like both getting a user's games AND doing the filtering) I start to think about how to split out the code into different modules. You could have a class that handles getting a user's games and a separate class for doing filtering on the games. That would make them a lot easier to reason about and easier to test as well, since each one has a distinct responsibility.

Reply:
That totally makes sense, thanks!